synth.py

```diff
@@ -5,24 +5,32 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
-gapic = gcp.GAPICGenerator()
+gapic = gcp.GAPICMicrogenerator()
 version = 'v1'
-library = gapic.node_library(
+library = gapic.typescript_library(
     'datastore', version,
-    config_path='/google/datastore/artman_datastore.yaml')
+    generator_args={
+        "grpc-service-config": f"google/datastore/{version}/datastore_grpc_service_config.json",
+        "package-name": f"@google-cloud/datastore",
+        "main-service": f"datastore"
+        },
+        proto_path=f'/google/datastore/{version}',
+        extra_proto_files=['google/cloud/common_resources.proto'],
+    )
 
 # Copy everything except for top level index, package.json, and README.md
 s.copy(
     library,
-    excludes=['package.json', 'README.md', 'src/index.js'])
-
-# Update path discovery due to build/ dir and TypeScript conversion.
-s.replace("src/v1/datastore_client.js", "../../package.json", "../../../package.json")
+    excludes=['package.json', 'README.md', 'src/index.ts'])
 
+system_test_files=['system-test/fixtures/sample/src/index.ts','system-test/fixtures/sample/src/index.js']
```

Comment:
I don't think we want this here ...  the system test isn't generate (at least I don't think).

Reply:
We do have really simple generated system-test from micro-generator using `packnplay` module, it just creates a new client for ts/js users. And since we have a manual layer for this client library, the client name gets changed from `DatastoreClient` -> `Datastore`. That's why here's the hack :(

Reply:
Yar!  My thinking - maybe we should make the emitting of the system test a feature behind a flag in the gapic generator, then set it to disabled for these bacon wrapped gapics?

Reply:
That's a good idea @JustinBeckwith, even though it's a little bit harder to implement (because of how the template engine works in the generator). I filed https://github.com/googleapis/gapic-generator-typescript/issues/320, we'll see what we can do.

Reply:
👍